modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ConcurrentLaws.kt

```diff
@@ -696,4 +697,19 @@ object ConcurrentLaws {
         startLatch.get().bind() toT onErrorRun.get().bind()
       }.equalUnderTheLaw(just(i toT false), EQ)
     }
+
+  fun <F> Concurrent<F>.outerAndInnerOnErrorIsRun(EQ: Eq<Kind<F, Unit>>, ctx: CoroutineContext) =
+    fx.concurrent {
+      val semaphore = Semaphore(0).bind()
```

Comment:
NIT: I'd rather it was an atomic count that we mutate, and check the count in the just.

Reply:
Up to you, otherwise tell me to merge and I'll do it.

Reply:
@pakoito please have another look

Reply:
LGTM. Merging as soon as the build finishes.

Reply:
there it is!